src/screens/SendTokens.js

```diff
@@ -182,6 +192,15 @@ function SendTokens() {
       data['inputs'] = [...data['inputs'], ...dataOne['inputs']];
       data['outputs'] = [...data['outputs'], ...dataOne['outputs']];
     }
+    // if there is any data output, add it to transaction's outputs
+    if (dataOutputs.length) {
+      dataOutputs.forEach((uId) => {
+        data.outputs.push({
+          type: OutputType.DATA,
+          data: String(dataOutputRefs.current[uId].current.value),
+        });
+      });
+    }
```

Comment:
Just to be clear on how this will work, the outputs array will have `[...tokenOutputs, ...dataOutputs]` meaning that conventionally all data outputs will be at the end of the outputs array.
If there is a change output added later the outputs will be shuffled.

This way works for me but if the intention is to have the outputs always at the end we should revisit how we organize the outputs.

Reply:
It's ok to shuffle the outputs in this case, no problem.